src/gromacs/nblib/bondtypes.h

```diff
@@ -0,0 +1,207 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with GROMACS; if not, see
+ * http://www.gnu.org/licenses, or write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA.
+ *
+ * If you want to redistribute modifications to GROMACS, please
+ * consider that scientific software is very special. Version
+ * control is crucial - bugs must be traceable. We will be happy to
+ * consider code for inclusion in the official distribution, but
+ * derived work must not be called official GROMACS. Details are found
+ * in the README & COPYING files - if they are missing, get the
+ * official version at http://www.gromacs.org.
+ *
+ * To help us fund GROMACS development, we humbly ask that you cite
+ * the research papers on the package. Check out http://www.gromacs.org.
+ */
+/*! \inpublicapi \file
+ * \brief
+ * Implements nblib simulation box
+ *
+ * \author Victor Holanda <victor.holanda@cscs.ch>
+ * \author Joe Jordan <ejjordan@kth.se>
+ * \author Prashanth Kanduri <kanduri@cscs.ch>
+ * \author Sebastian Keller <keller@cscs.ch>
+ * \author Artem Zhmurov <zhmurov@gmail.com>
+ */
+#ifndef GMX_NBLIB_BONDTYPES_H
+#define GMX_NBLIB_BONDTYPES_H
+
+#include "particletype.h"
+
+namespace nblib
+{
+using BondName      = std::string;
+using ForceConstant = real;
+using EquilDistance = real;
+using Exponent      = real;
+
+//! Harmonic bond type
+//
+// It represents the interaction of the form
+// V(r; forceConstant, equilDistance) = 0.5 * forceConstant * (r - equilDistance)^2
+class HarmonicBondType
+{
+public:
+    HarmonicBondType() = default;
+
+    HarmonicBondType(BondName bondName, ForceConstant forceConstant, EquilDistance equilDistance);
+
+    BondName bondName() const { return bondName_; }
+
+    ForceConstant forceConstant() const { return forceConstant_; }
+
+    EquilDistance equilDistance() const { return equilDistance_; }
+
+private:
+    BondName      bondName_;
+    ForceConstant forceConstant_;
+    EquilDistance equilDistance_;
+};
+
+
+//! GROMOS bond type
+//
+// It represents the interaction of the form
+// V(r; forceConstant, equilDistance) = 0.25 * forceConstant * (r^2 - equilDistance^2)^2
+class G96BondType
+{
+public:
+    G96BondType() = default;
+
+    G96BondType(BondName bondName, ForceConstant forceConstant, EquilDistance equilDistance);
+
+    BondName bondName() const { return bondName_; }
+
+    ForceConstant forceConstant() const { return forceConstant_; }
+
+    EquilDistance equilDistance() const { return equilDistance_; }
+
+private:
+    BondName      bondName_;
+    ForceConstant forceConstant_;
+    EquilDistance equilDistance_;
+};
+
+//! Cubic bond type
+//
+// It represents the interaction of the form
+// V(r; quadraticForceConstant, cubicForceConstant, equilDistance) = quadraticForceConstant * (r -
```

Comment:
This is awful. Shall we use more math-friendly names inside these classes? E.g.:
forceConstant -> ks
equilDistance -> r0...

Reply:
Verbose variable names are more user friendly.